### PR TITLE
refactor: creating suggestions and responses

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds response and suggestion method helpers to simplify and unify how users may work with responses and suggestions. 

Instead of using response and suggestion schemas init methods, users may use `*.with_question_value` method to create responses and suggestions given a question and a response value. This way will introduce some extra data validation which is not available using init methods directly.

For example, given the following dataset:
```python
dataset = rg.FeedbackDataset(
    fields =[rg.TextField(name="text")],
    question=[
        rg.TextQuestion(name="question-1", required=True),
        rg.RatingQuestion(name="question-2", values=[1, 2], required=True),
        rg.LabelQuestion(name="question-3", labels=["a", "b", "c"], required=True),
        rg.MultiLabelQuestion(name="question-4", labels=["a", "b", "c"], required=True),
        rg.RankingQuestion(name="question-5", values=["a", "b"], required=True),
    ]
)
```
users could create responses and suggestions as follows:

```python
question_1 = dataset.question_by_name("question-1")
question_2 = dataset.question_by_name("question-2")
question_3 = dataset.question_by_name("question-3")
question_4 = dataset.question_by_name("question-4")
question_5 = dataset.question_by_name("question-5")

record = rg.FeedbackRecord(
   fields={ "text": "This is a text value for field"},
   responses=[
      ResponseSchema(status="submitted")
        .with_question_value(question_1, value="answer")
        .with_question_value(question_2, value=0)
        .with_question_value(question_3, value="a")
        .with_question_value(question_4, value=["a", "b"])
        .with_question_value(question_5, value=[{"rank": 1, "value": "a"}, {"rank": 2, "value": "b"}])
   ],
   suggestions=[
      SuggestionSchema.with_question_value(question_1, value="answer"),
      SuggestionSchema.with_question_value(question_2, value=0),
      SuggestionSchema.with_question_value(question_3, value="a"),
      SuggestionSchema.with_question_value(question_4, value=["a", "b"]),
      SuggestionSchema.with_question_value(question_5, value=[{"rank": 1, "value": "a"}, {"rank":2, "value": "b"}])
   ]
)
```

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/schemas/suggestions.py [hunk 1]
```diff
@@ -59,3 +66,11 @@ def to_server_payload(self, question_name_to_id: Dict[str, UUID]) -> Dict[str, A
         payload["question_id"] = str(question_name_to_id[self.question_name])
 
         return payload
+
+    @classmethod
+    def with_question_value(cls, question: "QuestionSchema", value: ResponseValue, **kwargs) -> "SuggestionSchema":
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/schemas/suggestions.py]: maybe `from_question` is a better phrasing?

**Comment by davidberenstein1957** [comment]
Hi @frascuchon, this is a welcome addition. 

I might like the following better?

```python
question_1.to_suggestion(value="blabla)
question_1.to_response(value="blabla)
```

**Comment by frascuchon** [comment]
> Hi @frascuchon, this is a welcome addition.
> 
> I might like the following better?
> 
> ```python
> question_1.to_suggestion(value="blabla)
> question_1.to_response(value="blabla)
> ```

I see your point. Indeed I had a similar implementation approach at the beginning. My concern was the suggestion and response schemas work at a different level, since responses wrap all question values, the suggestions wrap a single question value.

Anyway, I pushed some changes with this idea. Now setting up responses and suggestions would be as follows:

```python
responses=[
    ResponseSchema(
        status="submitted",
        values=[
            question_1.response(value="answer"),
            question_2.response(value=0),
            question_3.response(value="a"),
            question_4.response(value=["a", "b"]),
            question_5.response(value=[{"rank": 1, "value": "a"}, {"rank": 2, "value": "b"}]),
        ],
    ),
  ]
  suggestions=[
      question_1.suggestion(value="answer"),
      question_2.suggestion(value=0),
      question_3.suggestion(value="a"),
      question_4.suggestion(value=["a", "b"]),
      question_5.suggestion(value=[{"rank": 1, "value": "a"}, {"rank": 2, "value": "b"}]),
  ]
````
As you can see it's quite similar but not the same. In any case, this can improve the usability of responses and suggestions. 

Let me know your thoughts.

**Comment by davidberenstein1957** [comment]
@frascuchon looks a lot better to me. Shouldn't we get rid of the `with_question_value` method now?

**Comment by frascuchon** [comment]
> Shouldn't we get rid of the `with_question_value` method now?

yep, If we go with the questions approach we should. I'll push the changes

**Pull request merged**